packages/ui/src/components/va-chip/VaChip.vue

```diff
@@ -35,139 +35,128 @@
+import { useColors, useColorProps } from '../../composables/useColor'
+import { useStateful, useStatefulEmits, useStatefulProps } from '../../composables/useStateful'
+import { useHover } from '../../composables/useHover'
 import VaIcon from '../va-icon'
 
-class ChipProps {
-  modelValue = prop<boolean>({ type: Boolean, default: true })
-  closeable = prop<boolean>({ type: Boolean, default: false })
-  color = prop<string>({ type: String, default: 'primary' })
-  outline = prop<boolean>({ type: Boolean, default: false })
-  flat = prop<boolean>({ type: Boolean, default: false })
-  icon = prop<string>({ type: String, default: '' })
-  disabled = prop<boolean>({ type: Boolean, default: false })
-  square = prop<boolean>({ type: Boolean, default: false })
-  tag = prop<string>({ type: String, default: 'span' })
-  size = prop<string>({
-    type: String,
-    default: 'medium',
-    validator: (value: string) => {
-      return ['medium', 'small', 'large'].includes(value)
-    },
-  })
-  shadow = prop<boolean>({ type: Boolean, default: false })
-}
-
-const ChipPropsMixin = Vue.with(ChipProps)
-
-@Options({
+export default defineComponent({
   name: 'VaChip',
-  components: { VaIcon },
-  emits: ['focus'],
-})
-export default class VaChip extends mixins(
-  RouterLinkMixin,
-  StatefulMixin,
-  ColorMixin,
-  ChipPropsMixin,
-) {
-  SetupContext = setup(() => {
-    const { hasKeyboardFocus, keyboardFocusListeners } = useKeyboardOnlyFocus()
 
-    return {
-      hasKeyboardFocus,
-      keyboardFocusListeners,
-    }
-  })
+  components: { VaIcon },
 
-  hoverState = false
-  focusState = false
+  emits: [...useStatefulEmits, 'focus'],
+
+  props: {
+    ...useRouterLinkProps,
+    ...useColorProps,
+    ...useStatefulProps,
+    modelValue: { type: Boolean, default: true },
+    closeable: { type: Boolean, default: false },
+    outline: { type: Boolean, default: false },
+    disabled: { type: Boolean, default: false },
+    square: { type: Boolean, default: false },
+    shadow: { type: Boolean, default: false },
+    flat: { type: Boolean, default: false },
+    icon: { type: String, default: '' },
+    tag: { type: String, default: 'span' },
+
+    size: {
+      type: String as PropType<'small' | 'medium' | 'large'>,
+      default: 'medium',
+      validator: (value: string) => ['small', 'medium', 'large'].includes(value),
+    },
+  },
 
-  created () {
-    watch(() => this.hoverState, (value) => {
-      this.updateHoverState(value)
-    })
-  }
+  setup (props, { emit }) {
+    const { getColor } = useColors()
+    const colorComputed = computed(() => getColor(props.color))
+    const borderColor = computed(() => props.outline ? colorComputed.value : '')
 
-  get iconSize () {
-    const size: any = {
+    const size = {
       small: '0.875rem',
       medium: '1rem',
       large: '1.25rem',
-    }
-    return size[this.size]
-  }
+    } as Record<string, string>
```

Comment:
```suggestion
    }
```

Reply:
Without `Record<string, string>` there is a typescript error for `size[props.size]`. Anyway, this part will be refactored (https://github.com/epicmaxco/vuestic-ui/issues/1671).